system-x/services/flink/src/main/java/software/tnb/flink/resource/openshift/OpenshiftFlinkOperator.java

```diff
@@ -0,0 +1,169 @@
+package software.tnb.flink.resource.openshift;
+
+import software.tnb.common.deployment.OpenshiftDeployable;
+import software.tnb.common.deployment.WithCustomResource;
+import software.tnb.common.deployment.WithExternalHostname;
+import software.tnb.common.deployment.WithName;
+import software.tnb.common.deployment.WithOperatorHub;
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.utils.HTTPUtils;
+import software.tnb.common.utils.WaitUtils;
+import software.tnb.flink.service.Flink;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.service.AutoService;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+import io.fabric8.kubernetes.api.model.GenericKubernetesResourceBuilder;
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodList;
+import io.fabric8.kubernetes.client.readiness.Readiness;
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.api.model.RouteBuilder;
+
+@AutoService(Flink.class)
+public class OpenshiftFlinkOperator extends Flink implements OpenshiftDeployable, WithExternalHostname, WithOperatorHub, WithCustomResource,
```

Comment:
same here, maybe this class should extend OpenshiftFlink instead of Flink

Reply:
Refactored with _OpenshiftFlink_ inheritance